### PR TITLE
some tech touchups (mainly clippy)

### PR DESCRIPTION
**Pull request by yonip23**
Hi all!
I started looking around in the source code while checking out [this issue](https://github.com/lapce/lapce/issues/603) I opened, and noticed that there are some compilation warnings, and there's no clippy running here.

So I took the liberty to:
1. Run `cargo clippy -- -D warnings` and fix all warnings
2. Fix some of Intellij's analysis warnings
3. Add clippy to the `ci.yml` and `pr.yml` workflows
4. Fix a small bug I detected, which is that the `ToggleTerminalFocus` was not available in the command palette - and now it is :-)
image

Hope you guys will find this PR beneficial 🙂

### CODE REVIEW

File: lapce-data/src/command.rs [hunk 1]
```diff
@@ -300,6 +300,7 @@ pub enum LapceWorkbenchCommand {
     TogglePanelVisual,
 
     // Focus toggle commands
+    #[strum(message = "Toggle Terminal Focus")]
```
Comment (yonip23) [inline lapce-data/src/command.rs:303]: this is the bugfix :-)

File: lapce-data/src/config.rs [hunk 2]
```diff
@@ -669,7 +669,7 @@ impl Config {
     }
 
     fn default_settings() -> config::Config {
-        config::Config::default()
+        Config::default()
```
Comment (dzhou121) [inline lapce-data/src/config.rs:672]: It is config::Config here.

Comment (yonip23) [reply lapce-data/src/config.rs:672]: What's the difference? I just removed what IJ told me that is an "unnecessary qualifier"

Comment (yonip23) [reply lapce-data/src/config.rs:672]: I see now that it's a false positive of JB 🙁 
When not specifying `config::Config`, the compiler goes to `lapce_data::config::Config` instead

**Comment by dzhou121** [comment]
Sorry can you fix the conflict and I'll merge.

**Comment by yonip23** [comment]
@dzhou121 I rebased
Thank you for reviewing 🙂

**Comment by yonip23** [comment]
@dzhou121 The master contained changes which failed the clippy check - I fixed them now, so please make sure the new fixes are ok before merging 🙂

**Comment by dzhou121** [comment]
Thank you!

**Pull request merged**